src/wayland.rs

```diff
@@ -154,3 +169,19 @@ fn gamma_control_cb(ctx: EventCtx<State, ZwlrGammaControlV1>) {
         _ => (),
     }
 }
+
+fn wl_output_cb(ctx: EventCtx<State, WlOutput>) {
+    if let wl_output::Event::Name(name) = ctx.event {
+        let i = ctx
+            .state
+            .outputs
+            .iter()
+            .position(|o| o.wl == ctx.proxy)
+            .unwrap();
+        let mut output = ctx.state.outputs.swap_remove(i);
+        let name = String::from_utf8(name.into_bytes()).expect("invalid output name");
+        eprintln!("Output {}: name = {name:?}", output.reg_name);
+        output.name = Some(name);
+        ctx.state.outputs.push(output);
```

Comment:
There is no need to take ownership of the output, so you can just `let output =  outputs.iter_mut().find(...).unwrap();`